assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -356,6 +368,15 @@ export default function ArchivePicker({view = ADMIN_VIEW}) {
     });
   };
 
+  const setCustomFeaturedImage = async imgId => {
+    await wp.data.dispatch('core/editor').editPost({featured_media: imgId});
+    dispatch({type: ACTIONS.CLOSE_SIDEBAR});
+    const closeBtn = document.querySelector('.media-modal-close');
+    if (closeBtn) {
+      closeBtn.click();
```

Comment:
From my testing this doesn't work, for me the modal only closes when setting a featured image that wasn't already imported to the library (so clicking on `Import to Library & Set Featured Image` instead of the `Set Featured Image`)